CMakeLists.txt

```diff
@@ -43,418 +50,536 @@ include("cmake/ontologies.cmake")
+		file(DOWNLOAD
+			https://raw.githubusercontent.com/jothepro/doxygen-awesome-css/v2.3.2/doxygen-awesome.css
+			${CMAKE_CURRENT_BINARY_DIR}/doxygen-awesome.css)
+	endif()
+
+	add_custom_target(doc
+		COMMAND ${DOXYGEN_EXECUTABLE} ${doxyfile}
+		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+		COMMENT "Generating API documentation with Doxygen"
+		VERBATIM)
+endif()
+
+# #############
+# ######### CPack
+# #############
 set(CPACK_GENERATOR "DEB")
 set(CPACK_PACKAGE_NAME ${PROJECT_NAME})
 set(CPACK_PACKAGE_VERSION ${PROJECT_VERSION})
-        set(CPACK_PACKAGE_DESCRIPTION ${PROJECT_DESCRIPTION})
-        set(CPACK_PACKAGE_HOMEPAGE_URL ${PROJECT_WEBSITE})
-        set(CPACK_DEBIAN_PACKAGE_MAINTAINER "Daniel Beßler <danielb@uni-bremen.de>")
+set(CPACK_PACKAGE_DESCRIPTION ${PROJECT_DESCRIPTION})
+set(CPACK_PACKAGE_HOMEPAGE_URL ${PROJECT_WEBSITE})
+set(CPACK_DEBIAN_PACKAGE_MAINTAINER "Daniel Beßler <danielb@uni-bremen.de>")
 set(CPACK_DEBIAN_PACKAGE_DEPENDS "libboost-program-options-dev, libboost-serialization-dev, libboost-python-dev, libeigen3-dev, libmongoc-dev, libraptor2-dev, librdf0-dev, libspdlog-dev, libgtest-dev, swi-prolog, python3-dev")
 set(CPACK_DEBIAN_PACKAGE_SECTION "devel")
 set(CPACK_DEBIAN_PACKAGE_ARCHITECTURE "amd64")
 set(CPACK_DEBIAN_PACKAGE_PRIORITY "optional")
 include(CPack)
 
-##############
-########## Unit Testings
-##############
+# #############
+# ######### Unit Testings
+# #############
 
 # add an executable target for GTest.
 # but the testing code is partly in the *knowrob* library
 # where gtest won't find them without using the "--no-as-needed"
 # flag for the linker.
 add_executable(all_gtests
-        tests/gtests.cpp
-        tests/QueryParserTest.cpp
-        tests/KnowledgeBaseTest.cpp
-        tests/StorageTest.cpp
-        tests/py/test-module.cpp
-        tests/TokenBroadcasterTest.cpp
-        tests/QueryTreeTest.cpp
-        tests/DependencyGraphTest.cpp
-        tests/DataSourceTest.cpp
-        tests/ConjunctiveBroadcasterTest.cpp
-        tests/UnifierTest.cpp
-        tests/py/test-jupyter.cpp
-        tests/py/PythonTests.cpp)
+	tests/gtests.cpp
+	tests/QueryParserTest.cpp
+	tests/KnowledgeBaseTest.cpp
+	tests/StorageTest.cpp
+	tests/py/test-module.cpp
+	tests/TokenBroadcasterTest.cpp
+	tests/QueryTreeTest.cpp
+	tests/DependencyGraphTest.cpp
+	tests/DataSourceTest.cpp
+	tests/ConjunctiveBroadcasterTest.cpp
+	tests/UnifierTest.cpp
+	tests/py/test-jupyter.cpp
+	tests/py/PythonTests.cpp)
 add_dependencies(all_gtests knowrob_py)
 
 # add sub-directory for plugin libraries
 add_subdirectory(tests/plugins)
 
 target_link_libraries(all_gtests
-        -Wl,--whole-archive,--no-as-needed
-        knowrob
-        -Wl,--no-whole-archive
-        ${Boost_Python_COMPONENT}
-        ${GTEST_MAIN_LIBRARIES})
-
+	-Wl,--whole-archive,--no-as-needed
+	knowrob
+	libswipl.so # Gtest needs this to find the Prolog engine
```

Comment:
this is not true for me locally, nor in CI! gtest runs just fine without

Reply:
without it I couldn't build the test for static libraries. In CI you have libswipl.so already in knowrob, that's why it works

Reply:
yes that should work normally. But because for some reason it needs some more functions in runtime :( for that the static libraries aren't enough :/